channeldb/channel.go

```diff
@@ -807,6 +811,20 @@ func (c *OpenChannel) MarkBorked() error {
 	return c.putChanStatus(ChanStatusBorked)
 }
 
+// isBorked returns true if the channel has been marked as borked in the
+// database. This requires an existing database transaction to already be
+// active.
+//
+// NOTE: The primary mutex should already be held before this method is called.
+func (c *OpenChannel) isBorked(chanBucket *bbolt.Bucket) (bool, error) {
```

Comment:
should we make this stricter, and be checking `isDefault` and disallowing state transitions in all other states?

Reply:
As is, the default state will always be set, and the remaining bits are interpreted as a bitvector.

Reply:
the default state is the all-zero case, i'm suggest making the check test if _any_ bits are present

Reply:
updated to disallow if non default (anything non default is borked)